extensions/minio/runtime/src/main/java/org/apache/camel/quarkus/component/minio/MinioRecorder.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.minio;
+
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.annotations.Recorder;
+import org.apache.camel.CamelContext;
+import org.apache.camel.quarkus.core.CamelContextCustomizer;
+
+@Recorder
+public class MinioRecorder {
+
+    public RuntimeValue<CamelContextCustomizer> createContextCustomizer(CamelMinioConfig minioConfig) {
+        return new RuntimeValue<>(new MinioContextCustomizer(minioConfig));
+    }
+
+    private static class MinioContextCustomizer implements CamelContextCustomizer {
+        private final CamelMinioConfig config;
+
+        public MinioContextCustomizer(CamelMinioConfig config) {
+            this.config = config;
+        }
+
+        @Override
+        public void customize(CamelContext camelContext) {
+            if (config.autowiredEnabled.isPresent())
+                camelContext.setAutowiredEnabled(config.autowiredEnabled.get());
```

Comment:
this would disable autowiring for any component, isn't it ?

Reply:
yes, now I understand what is wrong with it. I'll verify option to provide several clients into registry and change documentation accordingly (and remove this change, as it is not necessary)

Reply:
refactored so disabled autowiring is not suggested